Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -329,4 +392,32 @@ ScriptExecutionContext* CookieStore::scriptExecutionContext() const
     return ActiveDOMObject::scriptExecutionContext();
 }
 
+void CookieStore::eventListenersDidChange()
+{
+    bool hadChangeEventListener = m_hasChangeEventListener;
+    m_hasChangeEventListener = hasEventListeners(eventNames().changeEvent);
+
+    if (hadChangeEventListener == m_hasChangeEventListener)
+        return;
+
+    auto* context = scriptExecutionContext();
+    if (!context)
+        return;
+
+    auto& document = *downcast<Document>(context);
+    auto* page = document.page();
+    if (!page)
+        return;
+
+    auto& cookieJar = page->cookieJar();
```

Comment:
This function needs updating now that you store the cookie jar and host as data members.